tools/clang/lib/SPIRV/DeclResultIdMapper.cpp

```diff
@@ -3653,6 +3653,7 @@ SpirvVariable *DeclResultIdMapper::createSpirvStageVar(
   // According to DXIL spec, the SampleIndex SV can only be used by PSIn.
   // According to Vulkan spec, the SampleId BuiltIn can only be used in PSIn.
   case hlsl::Semantic::Kind::SampleIndex: {
+    setInterlockExecutionMode(spv::ExecutionMode::SampleInterlockOrderedEXT);
```

Comment:
Ok, here you are not setting the execution mode, just in case it requires this execution mode, limit it to SampleInterlockeOrderedExt.
Maybe the function should be named differently then.
What about something like `setAllowedInterlockExecutionMode` and by default any could be picked, unless we have this kind of restrictions?
Then the logic to pick the good on when you add the begin would look at the available ones and decide?

Reply:
`PixelInterlockOrderedEXT` should be the default execution mode for rasterizer ordered views

Reply:
yes, I meant like for the optional: this is a function on the base DeclResultIdMapper class. If I don't know about ROV, but see "setExecutionMode" for this attribute, I'm like "this attribute can ONLY be used with this mode".
But AFAIK SampleIndex SV don't require ROV. But if used within the scope of ROV, then it requires PixelInterlockOrderedEXT.
That's why I'm suggesting changing the name. So its clear it's just a restriction on the allowed execution mode for this interlock feature, but not a global restriction.

Reply:
Reading again, setInterlockExecutionMode is fine as a name. It scopes the thing enough.